lib/voxpupuli/test/rake/check.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'voxpupuli/test/rake/symlinks'
+
+namespace :check do
+  desc 'Fails if symlinks are present in directory'
+  task :symlinks do
+    symlinks = Voxpupuli::Test::Rake::Symlinks.new.check
+    unless symlinks.empty?
+      symlinks.each { |r| puts "Symlink found: #{r} => #{r.readlink}" }
+      raise 'Symlink(s) exist within this directory'
+    end
+  end
+
+  desc 'Fails if .pp files present in tests folder'
+  task :test_file do
+    ppfiles = Dir[File.join('tests', '**', '*.pp')]
+    unless ppfiles.empty?
+      puts ppfiles
+      raise '.pp files present in tests folder; Move them to an examples folder following the new convention'
+    end
+  end
+
+  desc 'Fails if any ._ files are present in directory'
+  task :dot_underscore do
+    dirs = Dir['._*']
+    unless dirs.empty?
+      puts dirs
+      raise '._ files are present in the directory'
+    end
+  end
+
+  desc 'Fails if directories contain the files specified in .gitignore'
+  task :git_ignore do
+    matched = `git ls-files --ignored --exclude-standard --cached`
+    raise 'git ls-files failed' unless $CHILD_STATUS.success?
+
+    unless matched == ''
+      puts matched
+      raise 'File specified in .gitignore has been committed'
+    end
+  end
+
+  desc 'Check for trailing whitespace'
+  task :trailing_whitespace do
+    errors = []
+
+    Dir.glob('**/*.md', File::FNM_DOTMATCH).sort.each do |filename|
+      next if filename =~ %r{^((modules|acceptance|\.?vendor|spec/fixtures|pkg)/|REFERENCE.md)}
+
+      File.foreach(filename).each_with_index do |line, index|
+        if line =~ /\s\n$/
```

Comment:
Was this ever correct? Shouldn't it be
```suggestion
        if line =~ /\s$/
```

And perhaps also change it to:
```suggestion
        if line.match?(/\s$/)
```